candle-wasm-examples/whisper/src/worker.rs

```diff
@@ -247,17 +300,44 @@ impl Decoder {
 
     pub fn load(md: ModelData) -> anyhow::Result<Self> {
         let device = Device::Cpu;
-        let tokenizer = Tokenizer::from_bytes(&md.tokenizer).map_err(anyhow::Error::msg)?;
+        let tokenizer = Tokenizer::from_bytes(&md.tokenizer).map_err(E::msg)?;
 
         let mel_filters = safetensors::tensor::SafeTensors::deserialize(&md.mel_filters)?;
         let mel_filters = mel_filters.tensor("mel_80")?.load(&device)?;
         console_log!("loaded mel filters {:?}", mel_filters.shape());
         let mel_filters = mel_filters.flatten_all()?.to_vec1::<f32>()?;
-        let vb = VarBuilder::from_buffered_safetensors(md.weights, DTYPE, &device)?;
-        let config = Config::tiny_en();
-        let whisper = Whisper::load(&vb, config)?;
+        let config: Config = serde_json::from_slice(&md.config)?;
+        let model = if md.quantized {
+            let vb = candle_transformers::quantized_var_builder::VarBuilder::from_gguf_buffer(
+                &md.weights,
+            )?;
+            Model::Quantized(m::quantized_model::Whisper::load(&vb, config)?)
+        } else {
+            let vb = VarBuilder::from_buffered_safetensors(md.weights, m::DTYPE, &device)?;
+            Model::Normal(m::model::Whisper::load(&vb, config)?)
+        };
         console_log!("done loading model");
-        let decoder = Self::new(whisper, tokenizer, mel_filters, &device)?;
+
+        let task = match md.task {
+            Some(task_str) => {
+                let task_string = task_str.to_owned();
+                match task_string.as_str() {
+                    "translate" => Some(Task::Translate),
+                    _ => Some(Task::Transcribe),
+                }
+            }
+            _ => Some(Task::Transcribe),
+        };
```

Comment:
is there a better way to to this? I want to send a `string` or None,  also tried to use an Enum with wasm_bindgen

Reply:
Maybe you could have a look at [this example](https://github.com/huggingface/candle/blob/11d3687cc655f8f79d856342a5539a9274e96df4/candle-examples/examples/quantized/main.rs#L306-L311), with the same`as_deref` thing you can probably do it with a single statement.